### PR TITLE
Support for most Corsair Peripherals (including RAM, Fans, LED Strips), added Shared Mode

### PR DESCRIPTION
**Pull request by Gurjot95**
**List any issues that this PR fixes:** fixes # , etc...
#1419 

**This pull request proposes the following changes:**

-  Adds support for most corsair peripherals such as RAM, Lighing Node, Commander Pro etc which follows mouse color
- Added device option to whether to run SDK in shared mode with iCue and option to disable each peripheral
- Optimized whole corsair device code (needs testing so all corsair devices are working fine)
- Updated SDK, this time for real

**Known issues/To do:**

- Needs extensive testing
- 
-

### CODE REVIEW

**Comment by Gurjot95** [comment]
Please make sure to inform me if this build is either working for you or not as I can then merge it to release build.

**Comment by Medroid** [comment]
Working for me. Tested with several profiles.

I have a commander pro with 4 rgb strips, a ST100 stand and a MM800 polaris mousepad. 

All working and using the same input as the mouse in aurora.

The only device not working/missing is the H110i logo. As image says below, it's using the SDK, but do not respond to the lighting effects.

image

**Comment by Gurjot95** [comment]
Cooler logo seems to be SDK issue as even offical sample fails to detect my device.

**Comment by srodriguez1850** [comment]
Corsair RAM doesn't light up when connected, sometimes doesn't connect to the SDK from the device manager or crashes when trying to.
Can test more if needed.

**Comment by zejjnt** [comment]
I have 3 ML fans, 2 LL fans, 2 sticks Vengeance RGB Pro, MM800 mousepad and a Razer Blackwidow keyboard.
The only thing that works correctly is the mousepad. The keyboard hangs at least once every ten minutes, often giving random error messages and force quitting Aurora (not just beta, has been like this for months).
The RAMs still use the same light pattern as they do when using iCUE normally, everything else is just off.
Please tell me how I can help troublershooting this.

**Comment by Lucasvor** [comment]
Gurjot95, you could add an Audio Visualizer to MM800 ? I have a corsair strafe, MM880, lighting node pro, with 3 HD 120 and 4 illumination node, they are all working perfectly with version Aurora-v0.6.4-584-bswuprap

**Comment by zejjnt** [comment]
> Gurjot95, you could add an Audio Visualizer to MM800 ? I have a corsair strafe, MM880, lighting node pro, with 3 HD 120 and 4 illumination node, they are all working perfectly with version Aurora-v0.6.4-584-bswuprap
Exactly the opposite for me, the only thing working is the MM800 :P

**Comment by Lucasvor** [comment]
you tried with this version: https://ci.appveyor.com/project/antonpup/aurora/builds/22008450/artifacts ?

**Comment by zejjnt** [comment]
> you tried with this version: https://ci.appveyor.com/project/antonpup/aurora/builds/22008450/artifacts ?

Fans do not light up, neither does the RAM. Only the mouse pad and the keyboard.

File: Project-Aurora/Project-Aurora/Devices/Corsair/CorsairDevice.cs [hunk 1]
```diff
@@ -381,41 +372,62 @@ private void SendColorToPeripheral(Color color, bool forced = false)
...
-                            mousemat[CorsairLedId.Zone14].Color = color;
-                        if (mousemat[CorsairLedId.Zone15] != null)
-                            mousemat[CorsairLedId.Zone15].Color = color;
-
-                        mousemat.Update(true);
-                    }*/
+                        lightingLED.Brush = (CUE.NET.Brushes.SolidColorBrush)color;
+                        lightingNode.Update(true);
+                    }
+                    if (memory != null && Global.Configuration.VarRegistry.GetVariable<bool>($"{devicename}_corsair_memory"))
```
Comment (srodriguez1850) [inline Project-Aurora/Project-Aurora/Devices/Corsair/CorsairDevice.cs]: `{devicename}_corsair_memory` should be `{devicename}_corsair_ram`, as set in the registry (L872). It's always returning false currently so memory modules don't light up.

Comment (Gurjot95) [reply Project-Aurora/Project-Aurora/Devices/Corsair/CorsairDevice.cs]: Thanks. Check #1530

**Comment by Gurjot95** [comment]
Closing this one as I have re-opened this on #1530

**Pull request closed without merging**